cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -40,24 +52,28 @@ func (c *CredentialsCmd) Run() error {
 	return err
 }
 
-var CLI struct {
+type CLI struct {
 	Credentials CredentialsCmd `kong:"cmd,help:'print credentials in a format usable as an AWS credentials_process'"`
 }
 
-func Run(args []string) error {
-	p, err := kong.New(&CLI)
```

Comment:
Although this worked, I don't think it was correct: we were using a "static" variable to record our parsing results into instead of just creating a fresh struct on every invocation.

Reply:
This often happens in Go where you declare a variable and it has a zero value (in our case an empty struct) and then pass it a pointer to update its value, e.g. whenever you deal with decoders (like a JSON decoder). I do agree with you, that avoiding global vars is a better practice.